prow/cmd/deck/main.go

```diff
@@ -1607,3 +1617,68 @@ func defaultLensRemoteConfig(lfc *config.LensFileConfig) error {
 
 	return nil
 }
+
+// ValidateStoragePath validates the storage bucket, if the `Deck.RestrictStoragePaths` config field is true.
+// The path should match a generic "<key-type>/<bucket>/<folder>/<sub-folder...>" pattern.
+// Additionally, the path bucket should match either the built-in and default storage locations _or_
+// should be contained in the `Deck.AdditionalAllowedBuckets` config field.
+// If the `Deck.RestrictStoragePaths` field is false, no validation errors are returned (but there will be warnings logged).
+func ValidateStoragePath(cfg config.Getter, path string) error {
```

Comment:
Why is this exported? It is not used outside of this package or did I miss that?

Reply:
Done. (No, it doesn't seem to be used outside. It has been lowerCamelCased.)